processor/src/lib.rs

```diff
@@ -78,18 +92,26 @@ struct Process {
 }
 
 impl Process {
-    pub fn new(inputs: ProgramInputs) -> Self {
-        Self {
+    fn initialize(inputs: ProgramInputs, enable_trace: bool) -> Process {
+        Process {
             system: System::new(4),
             decoder: Decoder::new(),
-            stack: Stack::new(&inputs, 4),
+            stack: Stack::new(&inputs, 4, enable_trace),
             hasher: Hasher::new(),
             bitwise: Bitwise::new(),
             memory: Memory::new(),
             advice: AdviceProvider::new(inputs),
         }
     }
```

Comment:
The constructors are simple enough - so, I'm not sure we need a separate function to encapsulate this. But I'm fine either way.

Reply:
Got it, just thought it would be easier as we have some magic numbers so avoid making mistakes.

Reply:
I'll leave this for now then